### PR TITLE
Extend From implementation for Tags

### PR DESCRIPTION
**Pull request by GuillaumeGomez**

### CODE REVIEW

File: src/objects.rs [hunk 1]
```diff
@@ -50,6 +50,12 @@ impl FromIterator<(String, String)> for Tags {
     }
 }
 
+impl<T: Iterator<Item = (String, String)>> From<T> for Tags {
+    fn from(f: T) -> Self {
+        FromIterator::from_iter(f)
+    }
+}
```
Comment (TeXitoi) [inline src/objects.rs]: That's not implemented for `Vec`, why? You already have '.collect()` that do the same.

**Comment by GuillaumeGomez** [comment]
You can't run with `X.filter(|x| true).into()` with just the `FromIterator` implementation.

**Comment by TeXitoi** [comment]
but you can `X.filter(|x| true).collect()`

**Comment by GuillaumeGomez** [comment]
Which isn't as nice since we need another `into_iter` inside flat_map.

**Comment by TeXitoi** [comment]
Example please

**Comment by GuillaumeGomez** [comment]
Sorry, I meant `tags::from_iter`.

**Comment by TeXitoi** [comment]
But just the fact that this kind of trait is not implemented in the collections in std is against adding such a trait.

Now, if I can have a better idea of what you are trying to improve, I might be more comprehensive.

**Comment by TeXitoi** [comment]
I don't get it. Please provide an example of before/after that you think your proposition will improve. I don't see why `.into()` is better than `.collect()` as it should be the only difference with this PR.

**Comment by GuillaumeGomez** [comment]
Maybe I'm not taking the right approach? Well, we can talk about it this evening if you're coming to the meetup. :)

**Comment by TeXitoi** [comment]
That's this evening :/ forgot. I'll try, but no promise, and no laptop.

**Comment by GuillaumeGomez** [comment]
No need for laptop!

**Comment by TeXitoi** [comment]
I'll be there.

**Comment by GuillaumeGomez** [comment]
I think you're right and this PR is mostly useless. But still: it allows to use `.into()` which is more common than `.collect()` (at least for me). If you don't think it's worth it, just close it. :)

**Comment by TeXitoi** [comment]
Collect is the canonical way of transforming an iterator into a collection.

Thanks for the proposition.

**Pull request closed without merging**